R/view_all_test_diff.R

```
@@ -0,0 +1,197 @@
+
+#' View Shinytest Diff
+#'
+#' @param dir Root app folder path
+#' @param sha git sha of base branch to look for
+#' @param ask Logical which allows for particular branches to be inspected
+#' @param ... Extra arguments passed to `shinytest::viewTestDiff`
+#' @param repo_dir Root repo folder path
+#' @export
+view_all_test_diff <- function(dir = "apps", sha = git_sha(dir), ask = interactive(), ..., repo_dir = file.path(dir, "..")) {
+  original_sys_call <- sys.call()
+  validate_core_pkgs()
+
+  validate_no_unexpected_shinytest_folders(dir)
+
+  branches <- gha_remotes_latest(dir = dir, sha = sha)
+  if (length(branches) == 0) {
+    message("Did not find any branches for sha: ", sha)
+    message("Be sure to run this function in your base branch. Current branch: ", git_branch(dir))
+    return()
+  }
+
+
+  if (isTRUE(ask)) {
+    ans <- utils::menu(c("(All branches)", branches), graphics = FALSE, title = "Select the Git branches you'd like to use")
+    # ans = 0; all
+    # ans = 1; all
+    if (ans > 1) {
+      # if ans is not 'all', subset the folders
+      ans_pos <- ans - 1
+      branches <- branches[ans_pos]
+    }
+  }
+
+  app_dir <- dir # must rename to avoid param error
+  git_cmd_ <- function(..., dir = app_dir) {
+    git_cmd(dir, paste0(...))
+  }
+  git_checkout <- function(git_branch_val) {
+    message("git checkout: ", git_branch_val)
+    git_cmd_("git checkout ", git_branch_val)
+  }
+
```

To help avoid any checkout failing weirdness, perhaps we should fail early if the original branch is in a "dirty" state (with a message to `git stash`)?